pkg/processor/sumologicsyslogprocessor/processor.go

```diff
@@ -0,0 +1,110 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package sumologicsyslogprocessor
+
+import (
+	"context"
+	"regexp"
+	"strconv"
+
+	"go.opentelemetry.io/collector/consumer/pdata"
+)
+
+// tailSamplingSpanProcessor handles the incoming trace data and uses the given sampling
+// policy to sample traces.
+type sumologicSyslogProcessor struct {
+	syslogFacilityAttrName string
+}
+
+const (
+	syslogSource = "syslog"
+)
+
+var (
+	facilities = map[int]string{
+		0:  "kernel messages",
+		1:  "user-level messages",
+		2:  "mail system",
+		3:  "system daemons",
+		4:  "security/authorization messages",
+		5:  "messages generated internally by syslogd",
+		6:  "line printer subsystem",
+		7:  "network news subsystem",
+		8:  "UUCP subsystem",
+		9:  "clock daemon",
+		10: "security/authorization messages",
+		11: "FTP daemon",
+		12: "NTP subsystem",
+		13: "log audit",
+		14: "log alert",
+		15: "clock daemon",
+		16: "local use 0  (local0)",
+		17: "local use 1  (local1)",
+		18: "local use 2  (local2)",
+		19: "local use 3  (local3)",
+		20: "local use 4  (local4)",
+		21: "local use 5  (local5)",
+		22: "local use 6  (local6)",
+		23: "local use 7  (local7)",
+	}
+)
+
+func newSumologicSyslogProcessor(cfg *Config) *sumologicSyslogProcessor {
+	return &sumologicSyslogProcessor{
+		syslogFacilityAttrName: cfg.FacilityAttr,
+	}
+}
+
+// ProcessLogs tries to extract facility number from log syslog line and maps it to facility name.
+// Facility is taken as $number/8 rounded down, where log looks like `^<$number> .*`
+func (ssp *sumologicSyslogProcessor) ProcessLogs(ctx context.Context, ld pdata.Logs) (pdata.Logs, error) {
+	// Iterate over ResourceLogs
+	rls := ld.ResourceLogs()
+	for i := 0; i < rls.Len(); i++ {
+		rl := rls.At(i)
+
+		ills := rl.InstrumentationLibraryLogs()
+		// iterate over InstrumentationLibraryLogs
+		for j := 0; j < ills.Len(); j++ {
+			ill := ills.At(j)
+
+			// iterate over Logs
+			logs := ill.Logs()
+			for k := 0; k < logs.Len(); k++ {
+				var (
+					value string = syslogSource
+					ok    bool
+				)
+
+				log := logs.At(k)
+				r, _ := regexp.Compile(`^<(?P<number>\d+)>`)
```

Comment:
This can be compile just once and then reused. Might be a separate PR + please check the err 😬